Formula/spago.rb

```diff
@@ -0,0 +1,23 @@
+class Spago < Formula
+  desc "PureScript package manager and build tool"
+  homepage "https://github.com/purescript/spago"
+  url "https://github.com/purescript/spago/archive/refs/tags/0.20.4.tar.gz"
+  sha256 "e2ef8604115556b39ec71301d85a3502502fd5972e40bac82cd556d6a128baff"
+  license "BSD-3-Clause"
+  head "https://github.com/purescript/spago.git", branch: "master"
+
+  depends_on "ghc" => :build
+  depends_on "haskell-stack" => :build
+  depends_on "purescript"
+
+  def install
+    system "make", "fetch-templates"
+    system "stack", "install", "--system-ghc", "--no-install-ghc", "--skip-ghc-check", "--local-bin-path=#{bin}"
+    (bash_completion/"spago").write `#{bin}/spago --bash-completion-script #{bin}/spago`
+    (zsh_completion/"_spago").write `#{bin}/spago --zsh-completion-script #{bin}/spago`
+  end
+
+  test do
+    assert_match "Spago - manage your PureScript projects", shell_output("#{bin}/spago --help")
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
I tried to write more substantial tests.  Please take a look again:

```ruby
assert_match "Spago - manage your PureScript projects", shell_output("#{bin}/spago --help")
testpath.cd do
  system bin/"spago", "init"
end
assert_predicate testpath/"packages.dhall", :exist?
assert_predicate testpath/"spago.dhall", :exist?
assert_predicate testpath/"src"/"Main.purs", :exist?
testpath.cd do
  system bin/"spago", "build"
end
assert_predicate testpath/"output"/"Main"/"index.js", :exist?
testpath.cd do
  assert_match "\u{1f35d}", shell_output("#{bin}/spago run")
end
```

Reply:
That's much better; thank you.